content/api/commands/selectfile.md

```diff
@@ -0,0 +1,276 @@
+[`cy.fixture`](/api/commands/fixture) or [`cy.readFile`](/api/commands/readfile)
+for more details on file encoding.
+
+### From an API response
+
+```javascript
+cy.request('http://localhost:8888/users/827').its('body').as('responseBody')
+
+cy.get('input[type=file]').selectFile('@responseBody')
+```
+
+### Processing data inside the test
+
+```javascript
+cy.readFile('users.json')
+  .then((users) => {
+    users[0].username = 'JohnCena'
+  })
+  .as('myFile')
+
+cy.get('input[type=file]').selectFile('@myFile')
+```
+
+### Attaching multiple files
+
+```javascript
+cy.get('input[type=file]').selectFile([
+  'file1.json',
+  'file2.json',
+  'file3.json',
+])
+```
+
+<Alert type="warning">
+
+This will fail unless the file input has the `multiple` property.
+
+</Alert>
+
+### Attaching a file with custom filename and lastModified
+
+```javascript
+cy.get('input[type=file]').selectFile({
+  contents: 'path/to/file.json',
+  fileName: 'users.json',
+  lastModified: new Date('Feb 18 1989').valueOf(),
+})
+```
+
+### Dropping a file on the document
+
+```javascript
+cy.document().selectFile('file.json', { action: 'drag-drop' })
+```
+
+## Notes
+
+### Existence
+
+#### Default file existence assertion
+
+Whenever resolving a file path, `.selectFile()` asserts that the file exists and
+will fail if it does not exist. It will retry reading the file if it does not
+initially exist until the file exists or the command times out.
+
+```javascript
+// will fail after the defaultCommandTimeout is reached
+cy.get('input[type=file]').selectFile('does-not-exist.yaml')
+```
+
+### Actionability
+
+#### The element must first reach actionability
+
+`.selectFile()` is an "action command" that follows all the rules of
+[Actionability](/guides/core-concepts/interacting-with-elements).
+
+## Rules
+
+### Requirements [<Icon name="question-circle"/>](/guides/core-concepts/introduction-to-cypress#Chains-of-Commands)
+
+- `.selectFile()` requires being chained off a command that yields DOM
+  element(s). With the `input` action (default), it further requires a single
+  `input` element with `type="file"`, or a `label` element attached to one.
+- If given a path, `.selectFile()` requires the file must exist.
+- If given an alias, `.selectFile()` requires that the subject of the alias must
+  not be `null` or `undefined`.
+
+### Assertions [<Icon name="question-circle"/>](/guides/core-concepts/introduction-to-cypress#Assertions)
+
+- `.selectFile()` will automatically wait for the element to reach an
+  [actionable state](/guides/core-concepts/interacting-with-elements).
+
+### Timeouts [<Icon name="question-circle"/>](/guides/core-concepts/introduction-to-cypress#Timeouts)
+
+- `.selectFile()` can time out waiting for the element to reach an
+  [actionable state](/guides/core-concepts/interacting-with-elements).
+- `.selectFile()` can time out waiting for a file to exist on disk or for an
+  alias to resolve.
+
+## Command Log
+
+**_Attach file to input_**
```

Comment:
```suggestion
**_Select file to upload to input_**
```

Reply:
Applying slightly different version of this one